### PR TITLE
handled enter event for empty h2/h3 tag 🔨

### LINKED ISSUE #924: event.preventDefault or disableDoubleReturn don't work when element is other than a 'p' tag
First and foremost, thank you for the excellent plugin!

I am getting issues when using 'disableDoubleReturn' or just a simple preventDefault on hitting Enter, when the node (line) is empty.

When the element is a simple 'p' tag, everything works as expected. No content = no new line created on "Return'. I need to write something inside the created line to be able to press Enter and create another one.

However, when I write something in the line, call up the toolbar and transform it into an 'h2' from a 'p', then delete everything in that 'h2' and press 'Return', the line returns as usually. Same for the 'h3'.

Is this an issue or wanted behaviour? If it's the latter, how would I go about overriding it? If it's the first, maybe someone can point me in the right direction to fix it?

Thank you in advance!

### PR DESCRIPTION
**Pull request by rishijain**
This fixes #924

### CODE REVIEW

File: src/js/core.js [hunk 1]
```diff
@@ -73,10 +73,12 @@
             } else if (MediumEditor.util.isKey(event, MediumEditor.util.keyCode.ENTER)) {
                 // hitting return in the begining of a header will create empty header elements before the current one
                 // instead, make "<p><br></p>" element, which are what happens if you hit return in an empty paragraph
-                p = this.options.ownerDocument.createElement('p');
-                p.innerHTML = '<br>';
-                node.previousElementSibling.parentNode.insertBefore(p, node);
-                event.preventDefault();
+                if (!this.options.disableDoubleReturn) {
```
Comment (nmielnik) [inline src/js/core.js]: Should this `!this.options.disableDoubleReturn` check be merged into the existiing outer-if?

ie:
```js
} else if (!this.options.disableDoubleReturn &&
              MediumEditor.util.isKey(event, MediumEditor.util.keyCode.ENTER)) {
    p = this.options.ownerDocument.createElement('p');
    // rest of code
}
              
```

File: spec/content.spec.js [hunk 2]
```diff
@@ -434,6 +434,45 @@ describe('Content TestCase', function () {
...
+            });
+
+            spyOn(evt, 'preventDefault').and.callThrough();
+
+            firePreparedEvent(evt, p, 'keydown');
+
+            expect(evt.preventDefault).not.toHaveBeenCalled();
+        });
+
+        it('should prevent new line being added when selected node is an empty h2/h3 tag and when disableDoubleReturn is true', function () {
```
Comment (nmielnik) [inline spec/content.spec.js]: Could you give these test cases different names/descriptions in some way?  Being named exactly the same thing can be a bit confusing...

Comment (j0k3r) [inline spec/content.spec.js]: @rishijain was it fixed when you force pushed your commit or I'm missing sth?

Comment (rishijain) [inline spec/content.spec.js]: @j0k3r Yes. I fixed those changes asked by @nmielnik  and then rebased and force pushed the commit as I do not want a separate commit just for fixing the typo.

Comment (j0k3r) [inline spec/content.spec.js]: Great, no problem :)

Comment (rishijain) [inline spec/content.spec.js]: I could have merged this already but I am waiting for @nmielnik  to have a quick look before it goes in. :smile:

Comment (j0k3r) [inline spec/content.spec.js]: Same here :D

**Comment by nmielnik** [comment]
:+1: looks good my man!
